### PR TITLE
HSEARCH-2816 Make sure Elasticsearch performance tests apply update/delete works to existing documents and add works to non-existing documents

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-2816

### CODE REVIEW

File: integrationtest/performance/engine-elasticsearch/src/main/java/org/hibernate/search/engineperformance/elasticsearch/ChangesetGenerator.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+		int updatesShuffledIdsOffset = deletesShuffledIdsOffset + addDeletesPerInvocation;
+		for ( int i = 0 ; i < addDeletesPerInvocation ; ++i ) {
+			toAdd.set( shuffledIds.get( i ) );
+		}
+		for ( int i = deletesShuffledIdsOffset ; i < updatesShuffledIdsOffset ; ++i ) {
+			toDelete.set( shuffledIds.get( i ) );
+		}
+		for ( int i = updatesShuffledIdsOffset ; i < threadIdIntervalSize ; ++i ) {
+			toUpdate.set( shuffledIds.get( i ) );
+		}
```
Comment (Sanne) [inline integrationtest/performance/engine-elasticsearch/src/main/java/org/hibernate/search/engineperformance/elasticsearch/ChangesetGenerator.java]: I'm not sure I understand the goal here. How do you guarantee that `toAdd` and `toDelete` are disjoint sets ?

Comment (yrodiere) [inline integrationtest/performance/engine-elasticsearch/src/main/java/org/hibernate/search/engineperformance/elasticsearch/ChangesetGenerator.java]: shuffledIds only contains unique indices, see how it's generated

Comment (Sanne) [inline integrationtest/performance/engine-elasticsearch/src/main/java/org/hibernate/search/engineperformance/elasticsearch/ChangesetGenerator.java]: oh, right. Looks good.

**Comment by Sanne** [comment]
merged this

**Pull request closed without merging**